book/typst/modules/module3.typ

```diff
@@ -1,8 +1,147 @@
+
+We can then create a system of differential equations to model the growth of both populations. 
+
+$ dif / (dif t) ("# ants at minute " t) 
+    &= a ("# ants at minute " t ) + b ("# aphids at time " t) \
+  dif / (dif t) ("# aphids at minute " t)
+    &= c ("# ants at minute " t ) - d ("# aphids at time " t) $
+
+The first equation models the growth of the ant population. The first term with parameter $a$ models the natural growth of ants in the absence of aphids. The second term with parameter $b$ models the growth of the ant population due to the presence of aphids -- this term provides a "boost" to the growth of the ant population since the presence of aphids will mean more food is available for the ants.
+
+The second equation models the growth of the aphid population. The first term with parameter $c$ models the growth of the aphid population due to the presence of ants -- this term has a positive effect to the growth of the aphid population since they will be protected from predators. The second term with parameter $d$ models the natural death of aphids in the absence of ants due to the fact that predation will overwhelm their natural growth.
+
+The parameters $a, b, c, d >= 0$ depend on the species of ants and aphids. As we discussed in MODULE 1, these constants can be estimated from data, but to do so, we need to be able to find the analytic solution to the system of differential equations. 
+In MODULES 6, 7, 8 we will learn how to solve some types of systems of differential equations like this one, called linear systems of differential equations with constant coefficients.
+
+
+
+== Simulation
+
+Just like we did in MODULE 2 to simulate one differential equation, we use Euler's method to simulate a system of differential equations.
+
+The method is the same as before, but now we have to keep track of two quantities at each step. We can do this by using a table to keep track of the values of the two populations at each time step.
+
+As an example, consider $a=1, b=c=d=1/2$ and starting populations of $10$ (thousand) ants and $100$ (thousand) aphids. We can simulate the system of differential equations using Euler's method with a time step of $0.25$:
+
+// #align(
+//         center,
+//         simple_table(
+//           headers: ([Time], [\# Ants], [\# Aphids]),
+//           content: (
+//           [0], [10], [100],
+//           [0.25], [25], [88.75],
+//           [0.5], [42.344], [80.781],
+//           [0.75], [63.027], [75.977],
+//           [1], [88.281], [74.358],
+//           [1.25], [119.646], [76.098],
+//           ),
+//         ),
+//       )
+
+#let a = 1
+#let b = 0.5
+#let c = 0.5
+#let d = 0.5
+#let F(x,y) = (a*x + b*y, c*x - d*y)
+#let v_0 = (10, 100)
+#let steps = 6
+#let Delta = 0.25
```

Comment:
Just a note about this, if you try to get `Δ` to show up in math, you'll have a hard time. I do `_delta` so the name doesn't clash.